AC_RUN_IFELSE is bad for cross compiling, switch to AC_COMPILE_IFELSE
since it's enough for this test.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff -Nura squid-3.3.9.orig/configure.ac squid-3.3.9/configure.ac
--- squid-3.3.9.orig/configure.ac	2013-09-13 09:15:32.473365627 -0300
+++ squid-3.3.9/configure.ac	2013-09-13 13:27:06.829711487 -0300
@@ -392,7 +392,7 @@
 dnl Check for atomic operations support in the compiler
 dnl
 AC_MSG_CHECKING([for GNU atomic operations support])
-AC_RUN_IFELSE([AC_LANG_PROGRAM([[
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
     int n = 0;
 ]],[[
     __sync_add_and_fetch(&n, 10); // n becomes 10
